### PR TITLE
feat(data): update validateEvent topic check

### PR DESCRIPTION
**Pull request by hahattan**
event topic from device service will be:
`<prefix>/<service-name>/<profile-name>/<device-name>/<source-name>`

corresponding to edgexfoundry/device-sdk-go#1261

Signed-off-by: Chris Hung <chris@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Comment by hahattan** [comment]
@lenny-intel according to document app-service expects event topic to have the same suffix:
https://docs.edgexfoundry.org/3.0/microservices/general/messagebus/#multi-level-topics-and-wildcards

However if event is added by REST API, coredata will not know the service name:
https://github.com/edgexfoundry/edgex-go/blob/main/internal/core/data/application/event.go#L84

Do we allow this inconsistency ? 
`edgex/events/device/<service>/<profile>/<device>/<source>` vs `edgex/events/core/<profile>/<device>/<source>`

or maybe we update the event topic published by coredata to `edgex/events/core/core-data/<profile>/<device>/<source>` ?

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
> However if event is added by REST API, coredata does not know the device service name:

@hahattan , the Add Event REST API needs to be update to require the `DeviceService`

**Comment by hahattan** [comment]
> > However if event is added by REST API, coredata does not know the device service name:
> 
> @hahattan , the Add Event REST API needs to be updated to include the `DeviceServiceName`

@lenny-intel REST endpoint only includes device name, profile name and source name.
Do we have any other way to retrieve device service name in this case ?

**Comment by cloudxxx8** [comment]
@hahattan after the discussion in the Core WG meeting on 15th-Feb-2023, we should modify the REST API to fit this change
[/event/{profileName}/{deviceName}/{sourceName}](https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.3.0#/default/post_event__profileName___deviceName___sourceName_)
it will become `/event/{serviceName}/{profileName}/{deviceName}/{sourceName}`

After this API change, you can get the device service name from REST API

**Comment by lenny-intel** [comment]
@hahattan , @cloudxxx8 , FYI may PR here https://github.com/edgexfoundry/edgex-go/pull/4348 took care of this except for updating the swagger. 

@cherrycl , FYI for TAF tests

File: internal/core/data/controller/messaging/subscriber.go [hunk 1]
```diff
@@ -114,11 +114,11 @@ func unmarshalPayload(envelope types.MessageEnvelope, target interface{}) error
 }
 
 func validateEvent(messageTopic string, e dtos.Event) errors.EdgeX {
-	// Parse messageTopic by the pattern `edgex/events/<device-profile-name>/<device-name>/<source-name>`
+	// Parse messageTopic by the pattern `edgex/events/device/<device-service-name>/<device-profile-name>/<device-name>/<source-name>`
 	fields := strings.Split(messageTopic, "/")
 
 	// assumes a non-empty base topic with /profileName/deviceName/sourceName appended by publisher
-	if len(fields) < 4 {
+	if len(fields) < 5 {
```
Comment (hahattan) [inline internal/core/data/controller/messaging/subscriber.go:121]: @lenny-intel Do you think we still need to check this ? I didn't see the change in #4348

Comment (cloudxxx8) [reply internal/core/data/controller/messaging/subscriber.go:121]: I believe we need this check, and update the comment in line 120
however, why doesn't it `if len(fields) < 6`?

**Pull request closed without merging**